packages/material-ui/src/Checkbox/Checkbox.js

```diff
@@ -136,9 +136,9 @@ Checkbox.propTypes = {
    */
   type: PropTypes.string,
   /**
-   * The value of the component.
+   * The value of the component. The DOM API casts this to a string.
    */
-  value: PropTypes.string,
+  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number, PropTypes.bool]),
```

Comment:
What do you guys think of using `.any` for all our input value types? I believe it's what React uses for a `<input>` element.

Reply:
Hmm, I think it's a good idea. Everything is cast to a string either way. As long as we're consistent :)

Reply:
cc @eps1lon

Reply:
The `Select` related components could be addressed here as well.

Reply:
I have just realized that this pull request makes the Checkbox component match the Radio and Switch value propType. I have noticed another inconsitancy.
`defaultChecked: PropTypes.bool,` doesn't match with `checked: PropTypes.oneOfType([PropTypes.bool, PropTypes.string]),`.

Reply:
TypeScript complains if you set defaultChecked, looking at the types both checked and defaultChecked should be boolean but then again value should be a string.
checked & defaultChecked https://github.com/Microsoft/TypeScript/blob/master/lib/lib.dom.d.ts#L6890
value https://github.com/Microsoft/TypeScript/blob/master/lib/lib.dom.d.ts#L7011

Reply:
Just change this to `any`. We continuously widened this in the past months anyway with all sorts of value components. Either we're strict or not. The arbitrary cuts that differ between every component just costs us time every other week.

Reply:
👍 for `any`.

Reply:
Hold one. The check prop types hasn't changed this the v1 rewrite: https://github.com/mui-org/material-ui/blame/35aa4e69047769149226a6e53ca856143eb76633/src/internal/SwitchBase.js. What do you think of using any for value but bool for checked?

Reply:
I was only talking about `value` (because that is already pretty wild in `Select` etc).

Reply:
Perfect, so everybody is aligned, we can proceed 🚀 .